addons/sale_pdf_quote_builder/models/sale_order_line.py

```diff
@@ -22,6 +22,16 @@ class SaleOrderLine(models.Model):
         readonly=False,
     )
 
+    # === ONCHANGE METHODS === #
+
+    @api.onchange('available_product_document_ids')
+    def _onchange_available_product_document_ids(self):
```

Comment:
or onchange product id instead ?
That way the onchange is not triggered everytime the form is opened (not fully sure it is but I think so)

Reply:
Hm, but then it'd be trigger when the product changed didn't change the available documents either (for instance, products without document change to another product without doc)... Or keep a document that was then deleted (and wouldn't be in the available any more too)